### ISSUE TITLE
Error: inject() must be called from an injection context

### ISSUE THREAD

**Issue by loicmarie** [dappsnation/akita-ng-fire#13]
I am using `@angular/cli@8.2.4`, but the error happens as well in `8.0.1` (for me)

To reproduce it, I just had to follow the steps in "Getting started" section.
After `ng serve`, the build runs well but the console outputs :
`Error: CollectionService requires AngularFirestore`

But I am injecting `AngularFirestoreModule` in AppModule, and I imported it in the service as well. My code does not differ from documentation, that is :

```
import { Injectable } from '@angular/core';
import { MovieStore, MovieState } from './movie.store';
import { AngularFirestore } from '@angular/fire/firestore';
import { CollectionConfig, CollectionService } from 'akita-ng-fire';

@Injectable({ providedIn: 'root' })
@CollectionConfig({ path: 'movies' })
export class MovieService extends CollectionService<MovieState> {

  constructor(store: MovieStore) {
    super(store);
  }

}
```

I tried to inject `AngularFirestore` service in constructor too, but it was not better.

So, I installed the project from source to output the real error and it was
`Error: inject() must be called from an injection context`
Seems linked to this - still open - [angular issue](https://github.com/angular/angular/issues/25813), but the proposed workaround does not work for me. I am having a hard time trying to understand this error.

**EDIT:** Here is how to easily reproduce it:
```
- ng new angular-test
- cd angular-test
- ng add @angular/fire
- ng add @datorama/akita
- ng g feature movies/movies
- (inject moviesService into app component constructor)
- ng serve
```

If necessary, I will add a stackblitz soon to reproduce this.

I was trying to do something similar to akita-ng-fire, but your project is way better. I would be glad to use it in our application, since you added atomic writings. This will be an essential tool for angularfire developers. Thumbs up :+1: 

NB: would be interesting to output the real error (or make a logger debug level) in addition to your error logs

**Comment by GrandSchtroumpf** [comment]
Hey, thanks for reporting this error. Sorry for the delay, I'll take a look asap. If you have any good practice recommandation for error handling, please don't hesitate to point me good ressources, or provide a PR.

**Comment by GrandSchtroumpf** [comment]
@loicmarie have you initialized your `AngularFireModule` as described [here](https://github.com/angular/angularfire2/blob/master/docs/install-and-setup.md#4-add-firebase-config-to-environments-variable) ?

**Comment by loicmarie** [comment]
Yes, it is initialized as described in the doc. The app uses `AngularFire`
 since months, without any problem.

I just removed all globally/locally linked librairies (linked with `npm link`) and now creating a new angular with `akita-ng-fire` app works well (I am not able to reproduce the issue with steps above anymore). But strangely, the big app is still not working. It's pretty hard to isolate the problem since the architecture is quite complex.

I am currently recreating the app from scratch, I will soon be able to say whether it was due to my dev environment or not. If it is, sorry for the trouble.

**Comment by GrandSchtroumpf** [comment]
You can try to delete `package.lock.json` and `node_modules` and run `npm install` again.
Also you may want to update `akita-ng-fire` to the latest version.

**Comment by loicmarie** [comment]
I already tried it, with no success. I don't known what exactly happened, but for the moment I don't have the problem anymore. I suspect some sensitive parts of our app of having broken the DI. I will close this issue, until the problem comes back (should not happen).

**Comment by loicmarie** [comment]
I finally found out what happened, that was dumb. When using linked local library, we always put the library in an external project. But we found out in [this issue](https://github.com/angular/angular-cli/issues/9273) that it is normally not working (don't know why it worked until here with our other old librairies), and so the `inject()` method was not working.

I just put the npm linked librairies into the `project/` directory and it's working like a charm now, sorry for having bothered you. I can send PR to `akita-ng-fire` now :)
